etna/models/tbats.py

```diff
@@ -17,27 +17,50 @@ class _TBATSAdapter(BaseAdapter):
     def __init__(self, model: Estimator):
         self.model = model
         self._fitted_model: Optional[Model] = None
+        self._last_train_timestamp = None
+        self._freq = None
 
     def fit(self, df: pd.DataFrame, regressors: Iterable[str]):
         target = df["target"]
         self._fitted_model = self.model.fit(target)
+        self._last_train_timestamp = df["timestamp"].max()
+        self._freq = pd.infer_freq(df["timestamp"], warn=False)
         return self
 
+    def _determine_num_steps_to_forecast(self, last_test_timestamp: pd.Timestamp) -> int:
+        diff = last_test_timestamp - self._last_train_timestamp
+        unit_diff = pd.timedelta_range(start=0, periods=2, freq=self._freq)[1]
+        num_steps = diff / unit_diff
+        return int(num_steps)
+
     def predict(self, df: pd.DataFrame, prediction_interval: bool, quantiles: Iterable[float]) -> pd.DataFrame:
         if self._fitted_model is None:
             raise ValueError("Model is not fitted! Fit the model before calling predict method!")
+
+        if df["timestamp"].min() <= self._last_train_timestamp:
```

Comment:
Let's rewrite message smth like: `in-sample predictions are not supported by current implementation`